### PR TITLE
Range chart unsupported queries

### PR DESCRIPTION
**Pull request by tyreer**
While working through a solution for the Range Chart, it became clear we could share some of the validation logic between visualizations so I created a utils folder and helper file for NRQL Validation. 

This PR: 
1. adds NRQL validation for the range chart. 
2. opts to have a single function for both aggregates & facets to keep it to a single loop over the data set for both, but it could be argued readability would win out and we could rewrite as two separate functions. 
3. creates a consistent way for us to determine the count of facets and aggregators across all visualizations. 
4. Moves the nrql validation function onto the class in the MultiFacet bar chart for consistency 
5. Updates nrql validation error language to be consistent

### CODE REVIEW

File: visualizations/range-chart/index.js [hunk 1]
```diff
@@ -114,6 +127,17 @@ export default class RangeChartVisualization extends React.Component {
                 return <ErrorState />;
               }
 
+              const isInputValid = this.validateNRQLInput(data);
+
+              if (!isInputValid) {
```
[the remark this answers is not in the document]
Comment (JuliaNocera) [reply visualizations/range-chart/index.js:122]: I like the rename to `nrqlInputIsValid` @tyreer - I assume we would make the same update across all components. Thanks for tending to these outstanding items

[the remark this answers is not in the document]
Comment (tyreer) [reply visualizations/range-chart/index.js]: Great, think we're good to go here then: 69dd1c3d2ce8a2477f7fb1c3dfa229521fc9e45e!

**Comment by jerelmiller** [comment]
Looks like there is a merge conflict + eslint issues. Once that is fixed, I'll do a full re-review.

**Comment by tyreer** [comment]
I'll have another look at the eslint issue, but will timebox and might ask for a hand. 

I spun my wheels on it a bit earlier.  I finding linting fails with what my format-on-save action picks up:

```
const { uniqueAggregates, uniqueFacets } = getUniqueAggregatesAndFacets(
    data
);
```

but also fails once you make the amend:

```
const { uniqueAggregates, uniqueFacets } = getUniqueAggregatesAndFacets(data);
```

Maybe we have competing linting rules?

image

image

**Review by jerelmiller — APPROVED** [review]
🔥 🔥 🔥

**Review by jerelmiller — APPROVED** [review]

**Pull request merged**